airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/cdc/MongoDbCdcInitializer.java

```diff
@@ -80,8 +79,11 @@ public List<AutoCloseableIterator<AirbyteMessage>> createCdcIterators(
                                                                         final Instant emittedAt,
                                                                         final MongoDbSourceConfig config) {
 
-    final Duration firstRecordWaitTime = RecordWaitTimeUtil.getFirstRecordWaitTime(config.rawConfig());
-    final Duration subsequentRecordWaitTime = RecordWaitTimeUtil.getSubsequentRecordWaitTime(config.rawConfig());
+    final Duration firstRecordWaitTime = Duration.ofSeconds(config.getInitialWaitingTimeSeconds());
+    // #35059: debezium heartbeats are not sent on the expected interval. this is
+    // a worksaround to allow making subsequent wait time configurable.
+    final Duration subsequentRecordWaitTime = firstRecordWaitTime;
```

Comment:
a max of 10 minutes may not be enough.
Customer is still seeing 101 records and timeout with 10 minutes.